### PR TITLE
Update c-blosc (v1) to 1.21.5

### LINKED ISSUE #429: Upgrade to ZSTD 1.5.5 due to potential corruption
https://github.com/Blosc/c-blosc/issues/364

https://github.com/facebook/zstd/releases/tag/v1.5.5

### PR DESCRIPTION
**Pull request by mkitti**
* Update c-blosc v1 to 1.21.5
* Fix #429 (upgrades Zstandard to 1.5.5)
* Compiles assembly files for x86_64 (.S files)
* Does not include minizip directory and thus does not confuse Python 3.8 header.

TODO:

- [x] Unit tests and/or doctests in docstrings
- [x] Tests pass locally
- [x] Docstrings and API docs for any new/modified user-facing classes and functions
- [x] Changes documented in docs/release.rst
- [x] Docs build locally
- [x] GitHub Actions CI passes
- [x] Test coverage to 100% (Codecov passes)

### CODE REVIEW

**Comment by mkitti** [comment]
```
/home/mkitti/review_temp/conda/3/x86_64/envs/zarrtest/compiler_compat/ld: build/temp.linux-x86_64-cpython-312/c-blosc/internal-complibs/zstd-1.5.5/decompress/huf_decompress.o: in function `HUF_decompress4X1_usingDTable_internal':
huf_decompress.c:(.text+0x3b86): undefined reference to `HUF_decompress4X1_usingDTable_internal_fast_asm_loop'
/home/mkitti/review_temp/conda/3/x86_64/envs/zarrtest/compiler_compat/ld: build/temp.linux-x86_64-cpython-312/c-blosc/internal-complibs/zstd-1.5.5/decompress/huf_decompress.o: relocation R_X86_64_PC32 against undefined hidden symbol `HUF_decompress4X1_usingDTable_internal_fast_asm_loop' can not be used when making a shared object
/home/mkitti/review_temp/conda/3/x86_64/envs/zarrtest/compiler_compat/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
[numcodecs] command '/usr/bin/gcc' failed with exit code 1
Traceback (most recent call last):
  File "/home/mkitti/review_temp/conda/3/x86_64/envs/zarrtest/lib/python3.12/site-packages/setuptools/_distutils/unixccompiler.py", line 266, in link
    self.spawn(linker + ld_args)
  File "/home/mkitti/review_temp/conda/3/x86_64/envs/zarrtest/lib/python3.12/site-packages/setuptools/_distutils/ccompiler.py", line 1041, in spawn
    spawn(cmd, dry_run=self.dry_run, **kwargs)
  File "/home/mkitti/review_temp/conda/3/x86_64/envs/zarrtest/lib/python3.12/site-packages/setuptools/_distutils/spawn.py", line 70, in spawn
    raise DistutilsExecError(
distutils.errors.DistutilsExecError: command '/usr/bin/gcc' failed with exit code 1
```

**Comment by joshmoore** [comment]
ICYMI: https://github.com/zarr-developers/numcodecs/pull/461

**Comment by mkitti** [comment]
This appears to be ready to go.

**Comment by mkitti** [comment]
Maybe I should not have skipped ci...

**Comment by d-v-b** [comment]
Anyone know what's blocking the github actions here?

**Comment by joshmoore** [comment]
No. It might be something with the orgs used resources.

File: docs/release.rst [hunk 1]
```diff
@@ -22,9 +22,15 @@ Fix
 
 * Fix skip of entry points backport tests
   By :user:`Elliott Sales de Andrade <QuLogic>`, :issue:`487`.
+* Fix Upgrade to Zstd 1.5.5 due to potential corruption
```
Comment (mkitti) [inline docs/release.rst:25]:
```suggestion
* Fix Upgrade to Zstd 1.5.5 due to potential corruption. 
```

**Comment by mkitti** [comment]
I added a period and did not skip CI this time. All tests pass. Let's upgrade Blosc1! It's been three years...

**Comment by d-v-b** [comment]
Any objections to getting this in?

**Pull request merged**